### PR TITLE
Add python3.10 in the ci configuration [build wheel linux] [build wheel osx] [build wheel win]

### PR DESCRIPTION
**Pull request by tshirtman**
Maintainer merge checklist
* [ ] Title is descriptive/clear for inclusion in release notes.
* [ ] Applied a `Component: xxx` label.
* [ ] Applied the `api-deprecation` or `api-break` label.
* [ ] Applied the `release-highlight` label to be highlighted in release notes.
* [ ] Added to the milestone version it was merged into.
* [ ] **Unittests** are included in PR.
* [ ] Properly documented, including `versionadded`, `versionchanged` as needed.

### CODE REVIEW

**Comment by misl6** [comment]
Here after the merge of #7678 😄 
Looks like we have some conflicts to solve.
I see an unrelated (should be now fixed) issue on `OSX Unittest`
Which is the status of this PR?

**Comment by tshirtman** [comment]
I guess kind of abandoned :laughing: ? Though I'm sure it can be resurrected, or replaced by a new one if it's easier.

**Comment by misl6** [comment]
Resurrected 😆

**Comment by misl6** [comment]
FYI:
I had to remove `Python 3.6` (which reached EOL) from the matrix, otherwise `Windows Unittests` were failing to run due to `ERROR: Package 'pyinstaller' requires a different Python: 3.6.8 not in '<3.11,>=3.7'` .

Maybe is a good sign to drop `Python 3.6` support (in a separate PR) 🧐 ?

**Comment by matham** [comment]
Yeah, we should drop it in a separate PR. This PR needs some working through the windows deps, I want to do it at some point not sure when. There's an upcoming sdl2 ttf release that [bundles harfbuzz](https://github.com/libsdl-org/SDL_ttf/issues/123), so we should probably wait for that!?

**Comment by tshirtman** [comment]
Oh, that sdl2_ttf with harfbuzz would be very welcome indeed. apparently there are pre releases package there https://www.libsdl.org/tmp/SDL_ttf/ in case you have time to experiment before that happens, but hopefully that release is not too far down the road.

**Comment by misl6** [comment]
@tshirtman @matham 

I think this PR is ready for a review:

✅ Windows wheels -> 3.7,  3.8, 3.9, 3.10
✅ Windows wheels tests -> 3.7,  3.8, 3.9, 3.10
✅ Manylinux wheels -> 3.7,  3.8, 3.9, 3.10
✅ Manylinux wheels tests -> 3.7, 3.8, 3.9, 3.10
✅ Migrated from Manylinux2010 to  Manylinux2014 (via cibuildwheel)
✅ x86_64 macOS wheel -> 3.7
✅ Universal2 macOS wheels ->  3.8, 3.9, 3.10
✅ x86_64 macOS wheel test -> 3.7
✅ Universal2 macOS wheel test (on x86_64) -> 3.8, 3.9, 3.10
✅ Universal2 macOS wheel test (on Apple Silicon M1 runner) -> 3.8, 3.9, 3.10

⚠️ Looks like raspberry PI wheels are only produced for cp37
⚠️ I only had a chance to locally test macOS wheels
⚠️ Wheels, are only tested in `dev` configuration, cause `ffpyplayer` does not provide (right now, we will switch back to `dev,full` in future) nor `universal2` wheels or `cp310` wheels.

File: .ci/ubuntu_ci.sh [hunk 1]
```diff
@@ -158,15 +162,46 @@ upload_docs_to_server() {
...
+  cd SDL2_image-2.0.4;
+  PATH="$HOME/kivy_build/bin:$PATH" PKG_CONFIG_PATH="$HOME/kivy_build/lib/pkgconfig" ./configure --prefix="$HOME/kivy_build" --bindir="$HOME/kivy_build/bin" --enable-png-shared=no --enable-jpg-shared=no --enable-tif-shared=no --enable-webp-shared=no;
+  PATH="$HOME/kivy_build/bin:$PATH" make;
+  make install;
+  make distclean;
+
+  cd ~/kivy_sources;
+  wget http://www.libsdl.org/projects/SDL_ttf/release/SDL2_ttf-2.0.15.tar.gz;
+  tar xzf SDL2_ttf-2.0.15.tar.gz;
+  cd SDL2_ttf-2.0.15;
```
Comment (matham) [inline .ci/ubuntu_ci.sh:200]:
```suggestion
  ver="2.0.18"
  wget "https://github.com/libsdl-org/SDL_ttf/releases/download/release-$ver/SDL2_ttf-$ver.tar.gz"
  tar xzf "SDL2_ttf-$ver.tar.gz"
  cd "SDL2_ttf-$ver"
```

File: pyproject.toml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 [build-system]
 requires = [
     "setuptools", "wheel",
-    "cython>=0.24,<=0.29.21,!=0.27,!=0.27.2",
+    "cython>=0.24,<=0.29.24,!=0.27,!=0.27.2",
```
Comment (matham) [inline pyproject.toml:4]:
```suggestion
    "cython>=0.24,<=0.29.26,!=0.27,!=0.27.2",
```

File: setup.cfg [hunk 3]
```diff
@@ -7,9 +7,9 @@ logging-level=DEBUG
 
 [kivy]
 cython_min=0.24
-cython_max=0.29.21
+cython_max=0.29.24
```
Comment (matham) [inline setup.cfg:10]:
```suggestion
cython_max=0.29.26
```

**Review by matham — CHANGES_REQUESTED** [review]

**Comment by matham** [comment]
Github is having action issues, which is I think what is causing the problems. E.g. the ubuntu tests is failing at the `python3 get-pip.py --user` step, so I don't think it's due to the changes.

**Review by matham — APPROVED** [review]

**Comment by misl6** [comment]
@matham Thank you for taking care of the `sdl2` versions for Ubuntu + the win deps!

**Pull request merged**